flume-ng-doc/sphinx/FlumeUserGuide.rst

```diff
@@ -234,6 +234,23 @@ The original Flume terminal will output the event in a log message.
 
 Congratulations - you've successfully configured and deployed a Flume agent! Subsequent sections cover agent configuration in much more detail.
 
+Using environment variables in configuration files
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If we enable environment variable interpolation by adding `-DpropertiesImplementation=org.apache.flume.node.EnvVarResolverProperties`
+to our `bin/flume-ng` call, we can use environment variables in the config, such as::
+
+  a1.sources = r1
+  a1.sources.r1.type = netcat
+  a1.sources.r1.bind = 0.0.0.0
+  a1.sources.r1.port = ${NC_PORT}
+  a1.sources.r1.channels = c1
+
+So a complete command line start could look like::
+
+  $ NC_PORT=44444 bin/flume-ng agent --conf conf --conf-file example.conf --name a1 -Dflume.root.logger=INFO,console -DpropertiesImplementation=org.apache.flume.node.EnvVarResolverProperties
+
+
```

Comment:
I totally agree with 1&2. Thanks for those.
I'm pushing a commit soon to address the issues you have mentioned.

Regarding the "we" concern: we have a lot of "we"-s (sic :) ) in the doc: https://github.com/apache/flume/blob/trunk/flume-ng-doc/sphinx/FlumeUserGuide.rst#a-simple-example
We can either stay consistent with it or rephrase those using passive voice, etc.